### PR TITLE
Fix axis Bug in MKLDNN Softmax

### PR DESCRIPTION
**Pull request by xinyu-intel**
## Description ##
Apply CheckAxis to MKLDNN Softmax function in order to call MKLDNN correctly.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [X] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

@pengzhao-intel @TaoLv

### CODE REVIEW

File: src/operator/nn/softmax.cc [hunk 1]
```diff
@@ -38,10 +38,9 @@ static void SoftmaxComputeExCPU(const nnvm::NodeAttrs& attrs,
                                 const std::vector<NDArray>& inputs,
                                 const std::vector<OpReqType>& req,
                                 const std::vector<NDArray>& outputs) {
-  const SoftmaxParam& param = nnvm::get<SoftmaxParam>(attrs.parsed);
   // It seems MKLDNN softmax doesn't support training.
   // and it only supports non-negative axis.
```
Comment (TaoLv) [inline src/operator/nn/softmax.cc]: please also remove this comment.

Comment (TaoLv) [reply src/operator/nn/softmax.cc]: why mkldnn softmax doesn't support training? https://github.com/intel/mkl-dnn/blob/master/include/mkldnn.hpp#L2680

Comment (xinyu-intel) [reply src/operator/nn/softmax.cc]: We are not using the latest MKLDNN.

**Comment by xinyu-intel** [comment]
@szha unit tests with negative axis have been added and passed.

**Comment by TaoLv** [comment]
@szha what's the difference of softmax/log_softmax/SoftmaxActivation in mxnet? and which one is the most widely used by mxnet users? Seems only softmax is optimized here with mkldnn primitive.

**Comment by szha** [comment]
right, the SoftmaxActivation is an older operator and is not actively developed anymore. Going forward we are using softmax/log_softmax.

**Comment by TaoLv** [comment]
@szha same question for SoftmaxOutput operator? https://github.com/apache/incubator-mxnet/blob/master/src/operator/softmax_output.cc#L47
